.github/workflows/pytest.yml

```diff
@@ -0,0 +1,31 @@
+name: Tests with Pytest
```

Comment:
maybe redundant adding a pytest specific workflow? But we can add tests for different OSes and python versions in addition to the one run in the code coverage